pkg/readiness/object_tracker.go

```diff
@@ -197,7 +229,7 @@ func (t *objectTracker) Observe(o runtime.Object) {
 	}
 
 	// Track for future expectation.
-	t.seen[k] = struct{}{}
+	t.seen[k] = mutatedObjData(t.mutators...)
```

Comment:
We probably want to leave this as an empty struct to avoid unnecessary memory usage.

Reply:
See my comment in other version of this comment.